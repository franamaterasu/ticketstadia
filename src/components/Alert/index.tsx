type AlertProps = {
  message: string;
};

const Alert = ({ message }: AlertProps) => {
  return (
    <div className='bg-red-500 text-white p-4 rounded-lg'>
      <p>{message}</p>
    </div>
  );
};

export default Alert;
