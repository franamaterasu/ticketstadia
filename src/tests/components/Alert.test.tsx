import { describe, test, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import Alert from '../../components/Alert';

describe('<Alert/>', () => {
  test('should match with snapshot', () => {
    const { container } = render(
      <Alert message='Im baby try-hard keytar kombucha cardigan selvage' />
    );

    expect(container).toMatchSnapshot();
  });

  test('should render title and paragraph', () => {
    render(
      <Alert message='Im baby try-hard keytar kombucha cardigan selvage' />
    );

    const message = screen.getByText(
      'Im baby try-hard keytar kombucha cardigan selvage'
    );

    expect(message).toBeDefined();
  });
});
