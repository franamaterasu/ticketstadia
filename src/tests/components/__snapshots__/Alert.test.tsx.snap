// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Alert/> > should match with snapshot 1`] = `
<div>
  <div
    class="bg-red-500 text-white p-4 rounded-lg"
  >
    <p>
      Im baby try-hard keytar kombucha cardigan selvage
    </p>
  </div>
</div>
`;
